sdk/core/azure-core/src/main/java/com/azure/android/core/http/options/AzureClientOptions.java

```diff
@@ -0,0 +1,76 @@
+package com.azure.android.core.http.options;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.azure.android.core.http.ServiceClient;
+import com.azure.android.core.util.logging.ClientLogger;
+
+public class AzureClientOptions {
+    @NonNull
+    private String apiVersion;
+    @Nullable
+    private ClientLogger logger;
+    @NonNull
+    private TelemetryOptions telemetryOptions;
+    @NonNull
+    private TransportOptions transportOptions;
+
+    /**
+     * Options for configuring a {@link ServiceClient}.
+     *
+     * @param apiVersion       The API version of the service to invoke.
+     * @param logger           The {@link ClientLogger} to be used by the {@link ServiceClient}.
+     * @param telemetryOptions Options for configuring telemetry sent by the {@link ServiceClient}.
+     * @param transportOptions Options for configuring calls sent by the {@link ServiceClient}.
+     */
+    public AzureClientOptions(@NonNull String apiVersion,
+                              @Nullable ClientLogger logger,
+                              TelemetryOptions telemetryOptions,
+                              TransportOptions transportOptions) {
+        this.apiVersion = apiVersion;
+        this.logger = logger;
+        this.telemetryOptions = telemetryOptions == null ? new TelemetryOptions(false, null) : telemetryOptions;
```

Comment:
Let's have constructor overloads that set these default values rather than specifying them here.

Reply:
What about the `ClientLogger`? If it's nullable I assume we would have something like this:

* ``` javapublic AzureClientOptions(@NonNull String apiVersion, @Nullable ClientLogger logger)`
* `public AzureClientOptions(@NonNull String apiVersion,
                              @Nullable ClientLogger logger,
                              @NonNull TelemetryOptions telemetryOptions,
                              @NonNull TransportOptions transportOptions)`

Reply:
Those look good except I would make ClientLogger @NonNull. Initially we had a default for the logger in Core for iOS as well, but realized that was a bad idea because you need to specify the name of the service client class for tagging log lines. So now each service client MUST define its own default ClientLogger instance. It can still be @Nullable in the concrete client options class (e.g. StorageBlobClientOptions) but the base class in Core should require it to be @NonNull.